src/lib.rs

```diff
@@ -46,25 +60,40 @@ fn update(message: &Message, model: &Model) -> Model {
     match message {
         Message::Increment => model.increment(),
         Message::Decrement => model.decrement(),
+        Message::ChangeText(text) => model.change_text(text),
     }
 }
 
 fn view(model: &Model) -> Html<Message> {
     Html::div(
-        &[],
-        &[
-            Html::button(
-                &[Attribute::on_click(&Message::Decrement)],
-                &[Html::text("-")],
-            ),
-            Html::span(
-                &[Attribute::class("counter")],
-                &[Html::text(&model.counter.to_string())],
+        vec![],
```

Comment:
Changed from array literals (slices?) to `Vec`s, as it makes the memory management much easier. (The contents aren't all clone-able, so it's easier to move the values rather than give a reference and then clone/copy.)